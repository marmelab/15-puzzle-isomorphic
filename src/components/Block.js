import React from 'react';
import PropTypes from 'prop-types';

import Row from './Row';

import withLoader from './withLoader';

const Block = ({ children, title }) => (
    <Row>
        {children}
        {title && <h5 className="center">{title}</h5>}
    </Row>
);

Block.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.node,
        PropTypes.arrayOf(PropTypes.element),
        PropTypes.arrayOf(PropTypes.node),
    ]).isRequired,
    title: PropTypes.string,
};

Block.defaultProps = {
    title: null,
};

export default withLoader(Block);
