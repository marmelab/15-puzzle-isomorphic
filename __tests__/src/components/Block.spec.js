import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Block from '../../../src/components/Block';

describe('Block', () => {
    test('should render Block without error', () => {
        renderer.create(
            <Block>
                <div>Hello</div>
            </Block>,
        );
    });

    test('should render the Block correctly', () => {
        const component = renderer.create(
            <Block>
                <div>Hello</div>
            </Block>,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should render the title prop if passed', () => {
        const app = shallow(
            <Block title="title">
                <div>Hello</div>
            </Block>,
        );
        expect(app.find('h5').text()).toEqual('title');
    });

    test('should display a loader', () => {
        const app = shallow(
            <Block title="title" isLoading={true}>
                <div>Hello</div>
            </Block>,
        );

        expect(app.find('.preloader-wrapper').length).toEqual(1);
    });
});
