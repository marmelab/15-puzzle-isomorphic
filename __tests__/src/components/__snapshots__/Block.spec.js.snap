// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block should render the Block correctly 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <div>
      Hello
    </div>
  </div>
  
</div>
`;
